ray-curator/ray_curator/stages/deduplication/semantic/base.py

```diff
@@ -0,0 +1,372 @@
+        # K-means clustering parameters
+        max_iter: int = 300,
+        tol: float = 1e-4,
+        random_state: int = 42,
+        init: Literal["k-means||", "random"] | np.ndarray = "k-means||",
+        n_init: int | Literal["auto"] = 1,
+        oversampling_factor: float = 2.0,
+        max_samples_per_batch: int = 1 << 15,
+        # Pairwise similarity parameters
+        distance_metric: Literal["cosine", "l2"] = "cosine",
+        which_to_keep: Literal["hard", "easy", "random"] = "hard",
+        ranking_strategy: RankingStrategy | None = None,
+        pairwise_batch_size: int = 1024,
+        limit: int | None = None,
+        # Duplicate identification parameters (optional)
+        eps: float | None = None,
+        num_row_groups_hint: int | None = None,
+        # I/O and storage parameters
+        read_kwargs: dict[str, Any] | None = None,
+        write_kwargs: dict[str, Any] | None = None,
+        # Execution parameters
+        verbose: bool = True,
+    ):
+        """
+        Initialize the semantic deduplication pipeline.
+
+        Args:
+            input_path: Directory or list of directories containing input files with embeddings
+            output_path: Directory to write output files
+            n_clusters: Number of clusters for K-means
+            id_field: Name of the ID field in the data
+            embedding_field: Name of the embedding field in the data
+            embedding_dim: Embedding dimension (for memory estimation)
+            metadata_fields: List of metadata field names to preserve in output
+            input_filetype: Type of input files ("parquet" or "jsonl")
+            input_file_extensions: List of file extensions to process
+            max_iter: Maximum number of K-means iterations
+            tol: Tolerance for K-means convergence
+            random_state: Random seed for K-means
+            init: K-means initialization method
+            n_init: Number of K-means initializations
+            oversampling_factor: K-means++ oversampling factor
+            max_samples_per_batch: Max samples per batch for K-means
+            distance_metric: Distance metric for similarity ("cosine" or "l2")
+            which_to_keep: Strategy for ranking within clusters ("hard", "easy", "random")
+            ranking_strategy: Custom ranking strategy (overrides which_to_keep)
+            pairwise_batch_size: Batch size for pairwise similarity computation
+            limit: Limit number of documents to process (for debugging)
+            eps: Epsilon value for duplicate identification
+            num_row_groups_hint: Number of row groups hint for duplicate removal
+            read_kwargs: Keyword arguments for reading files (including storage_options)
+            write_kwargs: Keyword arguments for writing files (including storage_options)
+            verbose: Enable verbose output
+            clear_output: Clear output directory before running
```

Comment:
Nit: this docstring could be more readable if you split it up by section (with blank lines between sections), like how you did for the init.